sdk/aqueduct/integrations/s3_integration.py

```diff
@@ -45,26 +48,37 @@ def file(
                 all matched files and concatenate them into a single file.
                 2) a list of strings representing the file name. Note that in this case, we do not
                 accept directory names in the list.
+            artifact_type:
+                The expected type of the S3 files. The `ArtifactType` class in `enums.py` contains all
+                supported types, except for ArtifactType.UNTYPED. Note that when multiple files are
+                retrieved, they must have the same artifact type.
             format:
-                The format of the S3 files. We currently support JSON, CSV, and Parquet. Note that currently,
-                when multiple files are retrieved, these files must have the same format.
+                If the artifact type is ArtifactType.TABLE, the user has to specify the table format.
+                We currently support JSON, CSV, and Parquet. Note that when multiple files are retrieved,
+                they must have the same format.
+            merge:
+                If the artifact type is ArtifactType.TABLE, we can optionally merge multiple tables
+                into a single DataFrame if this flag is set to True.
             name:
                 Name of the query.
             description:
                 Description of the query.
 
         Returns:
-            TableArtifact representing the concatenated S3 Files.
+            Artifact or a tuple of artifacts representing the S3 Files.
         """
-        lowercased_format = format.lower()
-        if lowercased_format == S3FileFormat.CSV.value.lower():
-            format_enum = S3FileFormat.CSV
-        elif lowercased_format == S3FileFormat.JSON.value.lower():
-            format_enum = S3FileFormat.JSON
-        elif lowercased_format == S3FileFormat.PARQUET.value.lower():
-            format_enum = S3FileFormat.PARQUET
+        if format:
+            lowercased_format = format.lower()
+            if lowercased_format == S3TableFormat.CSV.value.lower():
+                format_enum = S3TableFormat.CSV
+            elif lowercased_format == S3TableFormat.JSON.value.lower():
+                format_enum = S3TableFormat.JSON
+            elif lowercased_format == S3TableFormat.PARQUET.value.lower():
+                format_enum = S3TableFormat.PARQUET
+            else:
+                raise Exception("Unsupport file format %s." % format)
         else:
-            raise Exception("Unsupport file format %s." % format)
+            format_enum = None
```

Comment:
Don't we have to check that the format type is not table here, since you *must* supply a format type for table artifacts.